src/modules/utm/hooks.ts

```diff
@@ -65,7 +66,12 @@ export function useInitializeUtm(): void {
       // Clear params from URL and redirect
       const cleanedParams = cleanUpParams(searchParams)
       if (cleanedParams) {
-        navigate({ pathname, search: searchParams.toString() }, { replace: true })
+        const newSearch = searchParams.toString()
+        if (hasQueryParamsOutOfHashbang) {
+          window.location.replace(newSearch ? `${pathname}?${newSearch}` : '/')
```

Comment:
Shouldn't it always replace the location with at least `pathname` to remove the original params?

I don't think it makes sense to keep the trash in the url like this example: 

Test #6
From: https://swap-dev-git-allow-query-params-out-of-hashbang-cowswap.vercel.app?other=parameter
To: https://swap-dev-git-allow-query-params-out-of-hashbang-cowswap.vercel.app/?other=parameter#/1/swap/WETH